Sources/SwiftyRequest/RestRequest.swift

```diff
@@ -789,37 +793,49 @@ public enum RestError: Error, CustomStringConvertible {
 
 // URL Session extension
 extension RestRequest: URLSessionDelegate {
-
+    
     /// URL session function to allow trusting certain URLs
     public func urlSession(_ session: URLSession, didReceive challenge: URLAuthenticationChallenge, completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
         let method = challenge.protectionSpace.authenticationMethod
         let host = challenge.protectionSpace.host
-
+        
         guard let url = URLComponents(string: self.url), let baseHost = url.host else {
             completionHandler(.performDefaultHandling, nil)
             return
         }
-
+        
         let warning = "Attempting to establish a secure connection; This is only supported by macOS 10.6 or higher. Resorting to default handling."
-
+        
         switch (method, host) {
         case (NSURLAuthenticationMethodServerTrust, baseHost):
             #if !os(Linux)
-            guard #available(iOS 3.0, macOS 10.6, *), let trust = challenge.protectionSpace.serverTrust else {
+                guard #available(iOS 3.0, macOS 10.6, *), let trust = challenge.protectionSpace.serverTrust else {
+                    Log.warning(warning)
+                    fallthrough
+                }
+                if let certificateData = NSData(contentsOfFile: Bundle.main.path(forResource: self.pinnedCertificateName, ofType: "der") ?? "") {
+                    if let serverCertificate = SecTrustGetCertificateAtIndex(trust, 0) {
+                        let serverCertificateData = SecCertificateCopyData(serverCertificate) as NSData
+                        if certificateData == serverCertificateData {
+                            Log.verbose("Certificate validation successful")
+                            completionHandler(.useCredential, URLCredential(trust: trust))
+                            return
+                        } else {
+                            completionHandler(.performDefaultHandling, nil)
```

Comment:
Shouldn't we return after this? Otherwise we'll call the completion handler twice.

Reply:
Fixed.